File: src/Application Specific/Classes/Entity Collections/Order/OrderProduct/OrderProductValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace OrderProducts
    {
        // make sure that contact object is valid
        public class OrderProductValidator : BaseValidator
        {
            public void Validate(OrderProduct oOrderProduct)
            {

                // make sure that contact object is valid
                if (oOrderProduct.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oOrderProduct.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (!(oOrderProduct.Quantity > 0))
                {
                    AddCriticalMessage("Quantity must be greater than 0");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Order/OrderProduct/OrderProductValidator.vb
Namespace Entity

Namespace OrderProducts

Public Class OrderProductValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oOrderProduct As OrderProduct)

        'make sure that contact object is valid
        If oOrderProduct.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oOrderProduct.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

                If Not oOrderProduct.Quantity > 0 Then
                    Me.AddCriticalMessage("Quantity must be greater than 0")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace
